core/src/main/java/com/decibeltx/studytracker/core/events/util/EntryTemplateActivityUtils.java

```diff
@@ -0,0 +1,35 @@
+package com.decibeltx.studytracker.core.events.util;
+
+import com.decibeltx.studytracker.core.model.Activity;
+import com.decibeltx.studytracker.core.model.Activity.Reference;
+import com.decibeltx.studytracker.core.model.EntryTemplate;
+import com.decibeltx.studytracker.core.model.EventType;
+import com.decibeltx.studytracker.core.model.User;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+public class EntryTemplateActivityUtils {
+
+    private static Activity getActivity(EntryTemplate entryTemplate, User triggeredBy, EventType eventType) {
+        Activity activity = new Activity();
+        activity.setReference(Reference.ENTRY_TEMPLATE);
+        activity.setReferenceId(entryTemplate.getId());
+        activity.setEventType(eventType);
+        activity.setDate(new Date());
+        activity.setUser(triggeredBy);
+        Map<String, Object> data = new HashMap<>();
+        data.put("entryTemplate", entryTemplate);
+        activity.setData(data);
```

Comment:
I do not know all use cases of 'data' property, but if it is intended to use as readonly I would suggest: 
```suggestion
        activity.setData(Collections.singletonMap("entryTemplate", entryTemplate));
```